appinventor/appengine/src/com/google/appinventor/client/actions/DeleteAction.java

```diff
@@ -32,6 +33,7 @@ public void execute() {
             List<Project> projectsToDelete = selectedProjects;
             for (ProjectFolder f : selectedFolders) {
               projectsToDelete.addAll(f.getNestedProjects());
+              deleteFolder(f); // Call deleteFolder for each selected folder
```

Comment:
This forloop is just collecting a list of all affected projects to delete. No projects are actually deleted until after the confirmation is accepted.

To get the folders to actually delete, it looks like you can place a line inside the `deleteConfirmation` if statement after the folder is removed:

```
for (ProjectFolder f : selectedFolders) {
    f.getParentFolder().removeChildFolder(f);
}
Ode.getInstance().getFolderManager().saveAllFolders();
```

The deleteFolder method is unneccessary.